custom/shared/contexts/CallProvider.js

```diff
@@ -124,14 +124,11 @@ export const CallProvider = ({
     setPreJoinNonAuthorized(requiresPermission && !token);
   }, [state, daily, token]);
 
-  useEffect(() => {
-    if (!daily) return;
-
-    if (cleanURLOnJoin)
-      daily.on('joined-meeting', () => router.replace(`/${room}`));
+  const handleOnJoinCleanUp = useCallback(() => {
+    if (cleanURLOnJoin) router.replace(`/${room}`);
```

Comment:
can you move this onto its own line (we typically only use single line if statements for returns). 
```
if (cleanURLOnJoin) {
    router.replace(`/${room}`);
}
```